packages/connect-wallet-modal/src/components/WalletsModal/components/ConnectWalletModal/ConnectWalletModal.tsx

```diff
@@ -52,27 +52,26 @@ export const ConnectWalletModal = ({
 
   const isWalletsToggleButtonShown = useRef(false);
 
-  const [walletsList, setWalletsList] = useState<WalletConnectorData[]>([]);
-  const [walletsListFull, setWalletsListFull] = useState<WalletConnectorData[]>(
+  const [walletsList, setWalletsList] = useState<WalletAdapterData[]>([]);
+  const [walletsListFull, setWalletsListFull] = useState<WalletAdapterData[]>(
     [],
   );
 
-  const getWalletsListFull = useCallback(async () => {
-    // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
-    // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
-    // and a loading indicator is not required for now.
-    const _walletsListFull = await sortWalletsList({
-      config,
-      walletConnectorsList,
-      walletsShown,
-      walletsPinned,
-    });
-    setWalletsListFull(_walletsListFull);
-  }, [config, walletConnectorsList, walletsPinned, walletsShown]);
-
   useEffect(() => {
-    void getWalletsListFull();
-  }, [getWalletsListFull]);
+    const fetch = async () => {
+      // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
+      // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
+      // and a loading indicator is not required for now.
+      const _walletsListFull = await sortWalletsList({
+        config,
+        walletConnectorsList,
+        walletsShown,
+        walletsPinned,
+      });
+      setWalletsListFull(_walletsListFull);
+    };
+    void fetch();
+  }, [config, walletConnectorsList, walletsPinned, walletsShown]);
 
   useEffect(() => {
```

Comment:
This useEffect seems to just useMemo, it takes outer state filter/transforms it. Feels like we can do `walletsList=useMemo(...)`

Reply:
It is async fetch, useMemo is sync so that's why it is effect, not memo